3rd-party/class.jetpack-amp-vimeo-shortcode.php

```diff
@@ -0,0 +1,133 @@
+<?php //phpcs:ignore WordPress.Files.FileName.InvalidClassFileName
+
+/**
+ * Makes the [vimeo] Jetpack shortcode AMP-compatible.
+ *
+ * @see https://github.com/ampproject/amp-wp/blob/ea9e6fb9d262e699ea64978794840ba9868715f6/includes/embeds/class-amp-vimeo-embed.php#L71
+ */
+class Jetpack_AMP_Vimeo_Shortcode {
+
+	/**
+	 * Add the shortcode filter.
+	 */
+	public static function init() {
+		add_filter( 'do_shortcode_tag', array( 'Jetpack_AMP_Vimeo_Shortcode', 'filter_shortcode' ), 10, 3 );
+	}
+
+	/**
+	 * Filters the Vimeo shortcode to be AMP-compatible.
+	 *
+	 * @param string $html The video player HTML.
+	 * @param string $shortcode_tag The shortcode's tag (name).
+	 * @param array  $attr The attributes of the shortcode.
+	 * @return string The filtered HTML.
+	 */
+	public static function filter_shortcode( $html, $shortcode_tag, $attr ) {
+		if ( ! Jetpack_AMP_Support::is_amp_request() || 'vimeo' !== $shortcode_tag ) {
```

Comment:
Whenever we use the `Jetpack_AMP_Support` class, we try to check if it exists first, as it may not always exist in all environments (such as WordPress.com, where this code will be synchronized).

Reply:
Good idea. It's applied in https://github.com/Automattic/jetpack/pull/13921/commits/8afd00617c8bc4fce986176f3f8cc56c120d14e7